curriculum/challenges/english/25-front-end-development/lab-contact-form/66f26c32ec6f90df01a44f60.md

```diff
@@ -0,0 +1,420 @@
+---
+id: 66f26c32ec6f90df01a44f60
+title: Build a Contact Form
+challengeType: 14
+dashedName: lab-contact-form
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should have a `div` with a class of `form-container` to hold your form.
+
+2. You should have a `form` element within the `.form-container` `div`.
+
+3. Within the `form` element, you should have the following elements and input fields:
+
+   - A `h2` element with some text.
+   - A text input field for the name with the `type` set to `text` and `id`, `name`, `required` attributes.
+   - An email input field with the `type` set to `email` and `id`, `name`, `required` attributes.
+   - A textarea for the message with `id`, `name`, and the `required` attribute.
+   - A `button` element with the text `"Submit"` and a `type` attribute of `submit`.
+
+4. Each `input` and `textarea` elements should have their corresponding `label` element with a `for` attribute set to the element's `id`.
```

Comment:
I used a `input, textarea` selector in my CSS, which will not pass the tests for these. I think we should make a slight adjustment - maybe either mention that "You should have an `input` selector..." and "...`textarea` selector..." or adjust the tests to allow for more flexibility. You could check the computed values and not the stylesheet code.

Reply:
Added flexibility.